lua/starfall/libs_sv/prop.lua

```diff
@@ -430,9 +458,48 @@ function props_library.createSent(pos, ang, class, frozen)
 		end
 
 		hookcall = "PlayerSpawnedVehicle"
-
+	elseif sent2 then
+		if ply ~= NULL then
+			if (scripted_ents.GetStored(class).t.AdminOnly and not ply:IsAdmin()) then SF.Throw("This sent is admin only!", 2) end
+			if gamemode.Call("PlayerSpawnSENT", ply, class) == false then SF.Throw("Another hook prevented the sent from spawning", 2) end
+		end
+		
+		local enttbl = {}
+		data = data or {}
+		
+		-- Apply data
+		for param, org in pairs(sent2[1]) do
+			local value = data[param]
+			
+			if value then
+				checkluatype(value, org[2])
+				
+				enttbl[org[1]] = value
+			else
+				enttbl[org[1]] = org[3]
+			end
+		end
+		
+		-- Supply additional data
+		enttbl.Data = enttbl
```

Comment:
This should be two different tables. The assignments below may overwrite some data.

Reply:
Data is a special case that passes the whole structure.
https://wiki.facepunch.com/gmod/duplicator.RegisterEntityClass

Reply:
Ok. Weird